core/src/main/java/nav/statusplattform/core/openingHours/ServicesUpTimeRenumerator.java

```diff
@@ -0,0 +1,126 @@
+        }
+
+        //checks if the date entries are greater than the current date
+        if (DateEntryFrom.isAfter(now()) || DateEntryTo.isAfter(now())) {
+            throw new IllegalStateException("Arguments for DateEntry cannot be greater than the current date");
+        }
+
+        //checks for hh:mm-hh:mm
+        if (openingHours == null) {
+            throw new IllegalStateException("The argument openingHours must have a format pattern of 'hh:mm-hh:mm'");
+        }
+
+        //Application is always down
+        if (openingHours.equals("00:00-00:00")) {
+            return 0;
+        }
+
+        //Application is always up
+        if (openingHours.equals("00:00-23:59")) {
+            openingHours = "00:00-23:59";
+        }
+
+        String[] openingHoursParts = openingHours.split("-");
+
+        //Obtain the time in hh:mm format
+        String[] ohStartTime = openingHoursParts[0].split(":"); //start time hours[0] and minutes[1]
+        String[] ohEndTime = openingHoursParts[1].split(":"); //end time hours[0] and minutes[1]
+
+        //Convert string to local date format to get  LocalDateFormat that can be used to obtain zone datetime
+        LocalTime oHStartTime = LocalTime.of(Integer.parseInt(ohStartTime[0]), Integer.parseInt(ohStartTime[1]));
+        LocalTime oHEndTime = LocalTime.of(Integer.parseInt(ohEndTime[0]), Integer.parseInt(ohEndTime[1]));
+
+        //Obtained zonedDateTime
+        ZonedDateTime from = ZonedDateTime.of(DateEntryFrom, oHStartTime, ZoneId.of("Europe/Oslo"));
+        ZonedDateTime to = ZonedDateTime.of(DateEntryTo, oHEndTime, ZoneId.of("Europe/Oslo"));
+
+        return (int) calculatePercentageUptime(serviceId, from, to);
+
+    }
+
+    public static long calculatePercentageUptime(UUID serviceId, ZonedDateTime from, ZonedDateTime to) {
+        // These records Have to be sorted in chronological order
+
+        List<RecordEntity> records = recordRepository.getRecordsInTimeSpan(serviceId, from, to);
+
+        long sumOfActualUptime = 0; //Represents the total uptime for service for example,  service status equals
+        long sumOfExpectedUptime = 0; //Represents the total time
+
+        RecordEntity previousRecord = records.getFirst();
+
+        //Sum up (A) all the time  service has been UP, and all the time service should have been up
+        for (RecordEntity currentRecord : records.subList(1, records.size())) {
+
+            // Get the summarized amount of minutes of uptime specified within the timespan that the opening hours rule(s) expect
+            //Obtains the time difference between two periods
+            long expectedOpeningHoursTimeSpan;
+
+            int allDaysBetween = (int) (ChronoUnit.DAYS.between(from, to) + 1);
+
+            expectedOpeningHoursTimeSpan = IntStream.range(0, allDaysBetween)
+                    .count() * getOpeningHoursInMinutes(from, to);
```

Comment:
A. Removed +1 as it is not needed. 

1. Name changed to totalOHWithinTimespan to indicate that it is totalling the Opening Hours times between previous and current records.
2. The days between calculation is changed so that it is based on the time between previous and current records.